app/src/main/java/hypr/a255bits/com/hypr/MainInteractor.kt

```diff
@@ -4,15 +4,63 @@ import android.content.Context
 import android.graphics.Bitmap
 import android.net.Uri
 import android.provider.MediaStore
+import android.util.SparseArray
+import com.google.android.gms.vision.Frame
+import com.google.android.gms.vision.face.Face
+import com.google.android.gms.vision.face.FaceDetector
+import org.jetbrains.anko.toast
+
 
 /**
  * Created by ted on 7/17/17.
  */
-class MainInteractor(val context: Context): MainMvp.interactor{
-    override fun uriToBitmap(imageLocation: Uri): Bitmap {
-        return MediaStore.Images.Media.getBitmap(context.contentResolver, imageLocation);
+class MainInteractor(val context: Context) : MainMvp.interactor {
+    val detector: FaceDetector by lazy {
+        FaceDetector.Builder(context)
+                .setTrackingEnabled(false)
+                .setLandmarkType(FaceDetector.ALL_LANDMARKS)
+                .build()
+    }
+
+    override fun getFacesFromBitmap(imageWithFaces: Bitmap, width: Int, height: Int, context: Context): MutableList<Bitmap> {
+        val faceLocations: SparseArray<Face>? = getFaceLocations(imageWithFaces, context)
+        val croppedFaces = getListOfFaces(faceLocations, imageWithFaces)
+        return croppedFaces
+    }
+
+    private fun getFaceLocations(imageWithFaces: Bitmap, context: Context): SparseArray<Face>? {
+        var locationOfFaces = SparseArray<Face>()
 
+        if (detector.isOperational) {
+            val frame = Frame.Builder().setBitmap(imageWithFaces).build()
+            locationOfFaces = detector.detect(frame)
+        } else {
+            context.toast(context.resources.getString(R.string.failed_face_detection))
+        }
+        return locationOfFaces
     }
 
+    private fun getListOfFaces(faceLocations: SparseArray<Face>?, imageWithFaces: Bitmap): MutableList<Bitmap> {
+        val croppedFaces = mutableListOf<Bitmap>()
+        val numOfFaces: Int = faceLocations?.size()!!
```

Comment:
Nice! Yeah, it pretty much just allows that function to throw a NullPointerException, but it should only be used if you're sure it won't throw one.